contrib/golang/filters/http/source/go/pkg/http/cgo_go122.go

```diff
@@ -0,0 +1,54 @@
+//go:build go1.22
+
+package http
+
+/*
+// This is a performance optimization.
+// The following noescape and nocallback directives are used to
+// prevent the Go compiler from allocating function parameters on the heap.
+
+#cgo noescape envoyGoFilterHttpCopyHeaders
+#cgo nocallback envoyGoFilterHttpCopyHeaders
```

Comment:
Will Go < 1.22 raise an error when handling the unknown #cgo? Otherwise, we can put the comment before the function def.

Reply:
yep, old Go will raise an error: "invalid #cgo line".